staging/src/k8s.io/apiserver/pkg/registry/generic/registry/storage_factory.go

```diff
@@ -53,16 +53,16 @@ func StorageWithCacher(capacity int) generic.StorageDecorator {
 		// TODO: we would change this later to make storage always have cacher and hide low level KV layer inside.
 		// Currently it has two layers of same storage interface -- cacher and low level kv.
 		cacherConfig := cacherstorage.Config{
-			CacheCapacity:        capacity,
-			Storage:              s,
-			Versioner:            etcd3.APIObjectVersioner{},
-			ResourcePrefix:       resourcePrefix,
-			KeyFunc:              keyFunc,
-			NewFunc:              newFunc,
-			NewListFunc:          newListFunc,
-			GetAttrsFunc:         getAttrsFunc,
-			TriggerPublisherFunc: triggerFunc,
-			Codec:                storageConfig.Codec,
+			CacheCapacity:         capacity,
+			Storage:               s,
+			Versioner:             etcd3.APIObjectVersioner{},
+			ResourcePrefix:        resourcePrefix,
+			KeyFunc:               keyFunc,
+			NewFunc:               newFunc,
+			NewListFunc:           newListFunc,
+			GetAttrsFunc:          getAttrsFunc,
+			TriggerPublisherFuncs: triggerFuncs,
```

Comment:
do we plan to support indexing based on arbitrary fields ?

Reply:
no (at least not anytime in the near/mid future)